### PR TITLE
Add support for the `dnf5` package manager

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull requests adds support for the `dnf5` package manager.

## 💭 Motivation and context ##

We are adding support for Fedora 41 soon in cisagov/skeleton-ansible-role#207, and `ansible` identifies the package manager for that platform as `dnf5`, so we need to support it.

## 🧪 Testing ##

All automated tests pass.  These changes are also [tested](https://github.com/cisagov/freeipa-server-packer/actions/runs/11803640290) as part of cisagov/skeleton-ansible-role#207.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: tasks/dnf.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+- name: Configure dnf-automatic
+  block:
+    - name: Configure dnf-automatic to only consider security updates
```
Comment (mcdonnnj) [inline tasks/dnf.yml]: Would a loop make sense considering the overlap in all but the `option` and `value` arguments?

File: tasks/main.yml [hunk 2]
```diff
@@ -1,33 +1,34 @@
...
+- name: Load var file with package names based on the package manager
   ansible.builtin.include_vars:
     file: "{{ lookup('first_found', params) }}"
   vars:
     params:
       files:
-        - "{{ ansible_distribution }}_{{ ansible_distribution_release }}.yml"
-        - "{{ ansible_distribution }}.yml"
-        - "{{ ansible_os_family }}.yml"
+        - "{{ ansible_pkg_mgr }}.yml"
```
Comment (mcdonnnj) [inline tasks/main.yml:13]: Would it make sense to preemptively use the same configuration as the `include_tasks` below?

**Review by mcdonnnj — APPROVED** [review]
Just a couple items for your consideration.

**Review by dv4harr10 — APPROVED** [review]

File: tasks/dnf.yml (hunk 1 above)
Comment (jsf9k) [reply tasks/dnf.yml]: Please see commit [1fc5cdf](https://github.com/cisagov/ansible-role-automated-security-updates/pull/38/commits/1fc5cdf712e91f4fad1f9a6e91c28e10e1da7dc5).

File: tasks/main.yml (hunk 2 above)
Comment (jsf9k) [reply tasks/main.yml:13]: Please see commit [c0476df](https://github.com/cisagov/ansible-role-automated-security-updates/pull/38/commits/c0476dff422603e61e36fec76a1ad261f4bcbc0a).

**Review by dav3r — APPROVED** [review]
👍

**Pull request merged**